tests/Backend/Bigquery/TableDefinitionOperationsTest.php

```diff
@@ -810,4 +812,223 @@ public function testCreateSnapshotOnTypedTable(): void
+            'name' => 'my-new-table-for_data_preview',
+            'primaryKeysNames' => [],
+            'columns' => [
+                [
+                    'name' => 'column_int',
+                    'definition' => [
+                        'type' => 'INT64',
+                    ],
+                ],
+                [
+                    'name' => 'column_number',
+                    'definition' => [
+                        'type' => 'NUMERIC',
+                    ],
+                ],
+                [
+                    'name' => 'column_float',
+                    'definition' => [
+                        'type' => 'FLOAT64',
+                    ],
+                ],
+                [
+                    'name' => 'column_varchar',
+                    'definition' => [
+                        'type' => 'STRING',
+                    ],
+                ],
+                [
+                    'name' => 'column_datetime',
+                    'definition' => [
+                        'type' => 'DATETIME',
+                    ],
+                ],
+                [
+                    'name' => 'column_date',
+                    'definition' => [
+                        'type' => 'DATE',
+                    ],
+                ],
+                [
+                    'name' => 'column_time',
+                    'definition' => [
+                        'type' => 'TIME',
+                    ],
+                ],
+                [
+                    'name' => 'column_timestamp',
+                    'definition' => [
+                        'type' => 'TIMESTAMP',
+                    ],
+                ],
+                [
+                    'name' => 'column_boolean',
+                    'definition' => [
+                        'type' => 'BOOL',
+                    ],
+                ],
+            ],
+        ];
+
+        $csvFile = $this->createTempCsv();
+        $csvFile->writeRow([
+            'column_int',
+            'column_number',
+            'column_float',
+            'column_varchar',
+            'column_datetime',
+            'column_date',
+            'column_time',
+            'column_timestamp',
+            'column_boolean',
+        ]);
+        $csvFile->writeRow(
+            [
+                '1',
+                '3.14',
+                '3.14',
+                'roman',
+                '1989-08-31 00:00:00.000',
+                '1989-08-31',
+                '12:00:00.000',
+                '2023-04-18 12:34:56',
+                0,
+            ]
+        );
+
+        $tableId = $this->_client->createTableDefinition($bucketId, $tableDefinition);
+
+        $this->_client->writeTableAsync($tableId, $csvFile);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage($expectExceptionMessage);
+        $this->_client->getTableDataPreview($tableId, $params);
+    }
+
+    public function filterProvider(): Generator
```

Comment:
Je nějaký důvod proč mít ten filter provider teď 2x v každé třídě jednou, pokud to má 1:1 strukturu a víme, že to chceme zachovat 1:1? Vytahnul bych to třeba do traitu třeba. Akorát bych to tam asi přejmenoval na "wrongDatatypeFilterProvider" aby to bylo přesnější, když to bude reusable.